feed/feed-inbound/feed-inbound-1.0.0/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedInject.java

```diff
@@ -0,0 +1,116 @@
+    private final boolean sequential;
+    private final SynapseEnvironment synapseEnvironment;
+    private final String contentType;
+
+    public FeedInject(String injectingSeq, String onErrorSeq, boolean sequential,
+                      SynapseEnvironment synapseEnvironment, String contentType) {
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.sequential = sequential;
+        this.synapseEnvironment = synapseEnvironment;
+        this.contentType = contentType;
+    }
+
+    /**
+     * Determine the message builder to use, set the feed message to the
+     * message context and
+     * inject the message to the sequence
+     */
+    public boolean invoke(Object object) {
+        org.apache.synapse.MessageContext messageContext = null;
+        try {
+            messageContext = (org.apache.synapse.MessageContext) createMessageContext();
+            MessageContext axis2MsgCtx = null;
+            if (messageContext != null) {
+                axis2MsgCtx =
+                        ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+            }
+            // Determine the message builder to use
+            Builder builder = null;
+
+            if (axis2MsgCtx != null) {
+                builder = BuilderUtil.getBuilderFromSelector(contentType, axis2MsgCtx);
+            }
+            if (builder == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No message builder found for type '" + contentType + "'. Falling back to SOAP.");
+                }
+            }
+            OMElement documentElement = (OMElement) object;
+            messageContext.setEnvelope(TransportUtils.createSOAPEnvelope(documentElement));
+        } catch (AxisFault axisFault) {
+            log.error("Error while setting message to the message context :: " + axisFault.getMessage(), axisFault);
+        }
+        // Inject the message to the sequence.
+        if (StringUtils.isEmpty(injectingSeq)) {
+            log.error("Sequence name not specified. Sequence : " + injectingSeq);
+            throw new SynapseException("Sequence name not specified. Sequence : " + injectingSeq);
+        }
+        SequenceMediator seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(injectingSeq);
+        seq.setErrorHandler(onErrorSeq);
+        if (log.isDebugEnabled()) {
+            log.debug("injecting message to sequence : " + injectingSeq);
+        }
+        synapseEnvironment.injectInbound(messageContext, seq, sequential);
+        return true;
+    }
+
+    /**
+     * Create the initial messageContext for feed
+     */
+    private MessageContext createMessageContext() {
+        org.apache.synapse.MessageContext messageContext = synapseEnvironment.createMessageContext();
+        MessageContext axis2MsgCtx =
+                ((org.apache.synapse.core.axis2.Axis2MessageContext) messageContext).getAxis2MessageContext();
```

Comment:
Since you imported 'org.apache.synapse.core.axis2.Axis2MessageContext', you can use the class name only.